lib/hdprivatekey.js

```diff
@@ -106,8 +106,13 @@ HDPrivateKey._getDerivationIndexes = function(path) {
   }
 
   var indexes = steps.slice(1).map(function(step) {
-    var index = parseInt(step);
-    index += step != index.toString() ? HDPrivateKey.Hardened : 0;
+    var index = step ? +step : NaN;
+
+    var isHardened = isNaN(index) && step[step.length-1] == "'";
+    if (isHardened) {
+      index = (+(step.slice(0, -1))) + HDPrivateKey.Hardened;
+    }
+
     return index;
   });
 
```

Comment:
This code is way too complicated for what it does. I don't want to be a PITA, but what can you consider rewriting it in a simpler way? 
I'm talking about the whole `_getDerivationIndexes` function